### PR TITLE
EVG-7271 & EVG-7272: schedule and unschedule task mutations

### PR DESCRIPTION
**Pull request by SupaJoon**
This PR includes the `scheduleTask` and `unscheduleTask` mutations. The task reloaded from the db and sent back.
https://jira.mongodb.org/browse/EVG-7271
https://jira.mongodb.org/browse/EVG-7272

### CODE REVIEW

File: graphql/resolvers.go [hunk 1]
```diff
@@ -254,6 +254,52 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 	return testPointers, nil
 }
 
+func setSchedule(sc data.Connector, ctx context.Context, taskID string, isActive bool) (*restModel.APITask, error) {
```
Comment (tychoish) [inline graphql/resolvers.go]: always put the context as the first argument in a function.

**Comment by SupaJoon** [comment]
> I ran the generator with this schema and it made changes to the generated code. Could you rerun and commit that?

I reran the generator and didn't get any diffs

File: graphql/resolvers.go [hunk 2]
```diff
@@ -254,6 +254,52 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 	return testPointers, nil
 }
 
+func setSchedule(ctx context.Context, sc data.Connector, taskID string, isActive bool) (*restModel.APITask, error) {
```
Comment (john-m-liu) [inline graphql/resolvers.go]: nit: naming the function setScheduled would be more accurate

File: graphql/resolvers.go [hunk 3]
```diff
@@ -254,6 +254,52 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
...
+	usr := route.MustHaveUser(ctx)
+	if err := model.SetActiveState(taskID, usr.Username(), isActive); err != nil {
+		fmt.Println(err)
```
Comment (john-m-liu) [inline graphql/resolvers.go]: Left some debug code here

File: graphql/resolvers.go [hunk 4]
```diff
@@ -254,6 +254,52 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
...
+				"code": "INTERNAL_SERVER_ERROR",
+			},
+		}
+	}
+	task, err := task.FindOneId(taskID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	if task == nil {
+		return nil, errors.Errorf("unable to find task %s", taskID)
```
Comment (john-m-liu) [inline graphql/resolvers.go]: This should also be ResourceNotFound

File: graphql/resolvers.go [hunk 5]
```diff
@@ -254,6 +254,52 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 	return testPointers, nil
 }
 
+func setSchedule(ctx context.Context, sc data.Connector, taskID string, isActive bool) (*restModel.APITask, error) {
+	usr := route.MustHaveUser(ctx)
+	if err := model.SetActiveState(taskID, usr.Username(), isActive); err != nil {
+		fmt.Println(err)
+		return nil, &gqlerror.Error{
```
Comment (john-m-liu) [inline graphql/resolvers.go]: Why not use InternalServerError? Also I would pass through the error text as well

File: graphql/resolvers.go [hunk 6]
```diff
@@ -254,6 +254,46 @@ func (r *queryResolver) TaskTests(ctx context.Context, taskID string, sortCatego
 	return testPointers, nil
 }
 
+func setScheduled(ctx context.Context, sc data.Connector, taskID string, isActive bool) (*restModel.APITask, error) {
```
Comment (tgrander) [inline graphql/resolvers.go]: in future it might be helpful to put functions like these in a util folder since this file is already growing pretty big

**Pull request merged**